agent/src/containerutilityinterface.cpp

```diff
@@ -52,25 +52,29 @@ void ContainerUtilityInterface::removeOldContainers()
     log_warning() << num << " unused deprecated containers found";
     for (auto i = 0; i < num; i++) {
         struct lxc_container *container = containerList[i];
-        log_debug() << "Deprecated container named " << containerNames[i] << " will be deleted";
+        std::string containerName(containerNames[i]);
+        if (!std::regex_match(containerName, std::regex("SC-.*"))) {
+            log_debug() << "Found non-SC container " << containerName;
+            continue;
+        }
+
+        log_debug() << "Deprecated container named " << containerName << " will be deleted";
 
         if (container->is_running(container)) {
             bool success = container->stop(container);
             if (!success) {
-                std::string errorMsg = "Unable to stop deprecated container " +
-                                       std::string(containerNames[i]);
+                std::string errorMsg = "Unable to stop deprecated container " + containerName;
                 throw ContainerUtilityInterfaceError(errorMsg);
             }
         }
 
         bool success = container->destroy(container);
         if (!success) {
-            std::string errorMsg = "Unable to destroy deprecated container " +
-                                   std::string(containerNames[i]);
+            std::string errorMsg = "Unable to destroy deprecated container " + containerName;
             throw ContainerUtilityInterfaceError(errorMsg);
         }
 
-        log_debug() << "Deprecated container " << containerNames[i] << " is successfully destroyed";
+        log_debug() << "Deprecated container " << containerName << " is successfully destroyed";
         delete container;
         delete containerNames[i];
```

Comment:
sure but that does not mean that we need to keep the data stored that way right?